src/mavoscript.js

```diff
@@ -538,323 +520,11 @@ var _ = Mavo.Script = {
-					});
-				}
-				else {
-					comparisonOperands.forEach(co => {
-						ret.arguments.push(co.operand);
-					});
-				}
-			}
-			else {
-				// Flatten same operator calls
-				do {
-					ret.arguments.unshift(nodeLeft.right);
-					nodeLeft = nodeLeft.left;
-				} while (def.flatten !== false && nodeLeft.right && _.getOperatorName(nodeLeft.operator) === name);
-
-				ret.arguments.unshift(nodeLeft);
-			}
-
-			// Operator-specific transformations
-			def.postFlattenTransformation?.(ret);
-
-			return ret;
-		},
-		"UnaryExpression": node => {
-			var name = _.getOperatorName(node.operator, true);
-
-			if (name) {
-				return {
-					type: "CallExpression",
-					arguments: [node.argument],
-					callee: {type: "Identifier", name}
-				};
-			}
-		},
-		"CallExpression": node => {
-			if (node.callee.type == "Identifier") {
-				if (node.callee.name == "if") {
-					node.callee.name = "iff";
-
-					// Traverse data actions inside if() and rewrite them to their *if() counterpart
-					var condition = node.arguments[0];
-
-					for (let i=1; i<node.arguments.length; i++) {
-						if (i == 2) {
-							// Else, negate condition
-							condition = _.parse("not()");
-							condition.arguments.push(node.arguments[0]);
-						}
-
-						_.walk(node.arguments[i], n => {
-							var name = n.callee.name;
-
-							if (Mavo.Actions.Functions.hasOwnProperty(name) // is a data action
-							    && !/if$/.test(name) // and not already the *if() version of itself
-							) {
-								n.callee.name += "if";
-
-								// Add condition as first argument of *if() function
-								n.arguments.unshift(condition);
-							}
-						}, {type: "CallExpression"});
-					}
-				}
-				else if (node.callee.name == "delete") {
-					node.callee.name = "clear";
-				}
-				else {
-					var def = Mavo.Functions[node.callee.name];
-
-					if (def && def.needsContext) {
-						// Rewrite to funcName.call($this, ...args)
-						node.callee = {
-							type: "MemberExpression",
-							computed: false,
-							object: node.callee,
-							property: {type: "Identifier", name: "call"}
-						};
-						node.arguments.unshift({type: "Identifier", name: "$this"});
-					}
-				}
-			}
-		},
-		"ThisExpression": node => {
-			return {type: "Identifier", name: "$this"};
-		}
-	},
-
-	closest (node, type) {
-		let n = node;
-
-		do {
-			if (n.type === type) {
-				return n;
-			}
-		} while (n = n.parent);
-
-		return null;
-	},
-
-	serialize: (node, parent) => {
-		if (typeof node === "string") {
-			return node; // already serialized
-		}
-
-		if (parent) {
-			node.parent = parent;
-		}
-
-		var ret = _.transformations[node.type]?.(node, parent);
-
-		if (typeof ret == "object" && ret?.type) {
-			node = ret;
-		}
-		else if (ret !== undefined) {
-			return ret;
-		}
-
-		if (!node.type || !_.serializers[node.type]) {
-			throw new TypeError("Cannot understand this expression at all 😔");
-		}
-
-		return _.serializers[node.type](node, parent);
-	},
```

Comment:
Oooh that must have been pleasant 😁

However, we still need the transformations (vastly has an export of an empty transformations object so the developer can modify them). Keep in mind vastly is written for parsing/serializing regular JS, so all the special Mavo stuff will still live here.

Reply:
Try `Object.assign(Vastly.transformations, transformations)` but do check where rollup puts that export, it may be `Vastly.serialize.transformations` or something entirely different.

Reply:
Followup question: where should this go such that it only runs once? Somewhere around [here](https://github.com/mavoweb/mavo/blob/main/src/mavo.js#L8)?

Reply:
Update: `transformations` and other non-default exports are not accessible outside Vastly. I assume it's because the index file only exports the defaults for each file: https://github.com/mavoweb/vastly/blob/main/src/index.js#L2C1-L9C55

Reply:
> Update: `transformations` and other non-default exports are not accessible outside Vastly. I assume it's because the index file only exports the defaults for each file: [mavoweb/vastly@`main`/src/index.js#L2C1-L9C55](https://github.com/mavoweb/vastly/blob/main/src/index.js?rgh-link-date=2023-11-28T19%3A32%3A56Z#L2C1-L9C55)

That's something to fix on the vastly side :) Please file an issue first, so you have something to close.

One way is to add it as a property of `serialize`, another to add it as a separate named export. Which solution would you pick and why? :)

Reply:
Yeah I agree, I was asking where _in mavo_ should we put the line Object.assign(Vastly..., transformations). I suggested [here](https://github.com/mavoweb/mavo/blob/main/src/mavo.js#L8) since there's a very similar line right below to set something inside stretchy

Reply:
> That's something to fix on the vastly side :) Please file an issue first, so you have something to close.

Will do

> One way is to add it as a property of `serialize`, another to add it as a separate named export. Which solution would you pick and why? :)

I'm leaning toward adding it as a prop of `serialize`, because adding additional lines like `export {transformations} from "./serialize.js"` doesn't seem too good to me. When you say add as a property of serialize, how would I go about doing this?

Reply:
Ok, let me elaborate a bit, as I think my earlier message wasn't super clear.

Option 1 is that `index-fn` also does `export { transformations, serializers }`. Then they become top-level exports, same as the functions, and they'd be available as e.g. `Vastly.transformations`. 

Option 2 is that we add them as properties of the `serialize` function, since functions are objects in JS:

```
serialize.transformations = transformations;
```

Then they are not exported per se, but can be accessed via `serialize`.

What do you think are the pros & cons of each of these approaches?

Reply:
That's cool! To me, it seems like the pros of option1 is we don't waste space inside of serialize "object", and the pros of option2 is we don't have to worry about exporting a separate thing from index, and `transformations` is tied to `serialize` since that's where it belongs